### PR TITLE
Feature/ver 72

### PR DESCRIPTION
**Pull request by Ellenn-A**
- added folder 'definitions' with a schema definition in it (please let me know if this folder should 'sit' somewhere else in hierarchy) 
- added tests for creating a schema and fetching it, using the new schema definition
- added tests for creating credential definition and fetching it, using new schema definition

Question: I have not removed any tests - should I have done that?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@digicatapult/veritable-cloudagent",
-  "version": "0.3.1",
+  "version": "0.4.1",
```
Comment (jonmattgray) [inline package.json:3]: `0.4.0`

File: definitions/schemaDefinition.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (jonmattgray) [inline definitions/schemaDefinition.json:1]: could add something to the `README` explaining what this file is and how to use it

File: tests/credentialDefinition.test.ts [hunk 3]
```diff
@@ -82,12 +85,55 @@ describe('CredentialDefinitionController', () => {
...
     test('should return 404 NotFound when credential definition not found', async () => {
+      const spy = stub(agent.modules.anoncreds, 'getCredentialDefinition')
+      spy.resolves({
+        credentialDefinitionId: 'x',
+        credentialDefinitionMetadata: {},
+        resolutionMetadata: {
+          error: 'notFound',
+        },
+      })
+
```
Comment (jonmattgray) [inline tests/credentialDefinition.test.ts:96]: how was the test passing before, without the stub?

Comment (Ellenn-A) [reply tests/credentialDefinition.test.ts:96]: It wasn't? - I just ran tests on main and it failed there too.

Comment (jonmattgray) [reply tests/credentialDefinition.test.ts:96]: weird, idk how that got into `main`. Good that this fixes it anyway

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: tests/schema.test.ts [hunk 4]
```diff
@@ -59,7 +61,7 @@ describe('SchemaController', () => {
         schemaMetadata: {},
       })
 
-      const response = await request(app).get(`/schemas/x`)
+      const response = await request(app).get(`/schemas/x`) //inspo
```
Comment (jonmattgray) [inline tests/schema.test.ts:64]: remove comment?

File: definitions/schemaDefinition.json (hunk 2 above)
Comment (Ellenn-A) [reply definitions/schemaDefinition.json:1]: Can I add this? 



### Schema Definition

The repo contains 'definitions' folder with a schema definition json which can be imported into ts files like so:

```
import _schema from '<path>/definitions/schemaDefinition.json'. 
const schema = _schema as AnonCredsSchema
```

The schema definition can be posted to `/schemas` to register it. Upon a successful call a response is returned with an 'id' property which refers to this schema and can be used to refer to the schema when creating a new credential definition.
(Note: when registering a schema one needs to have an instance of ipfs running or be connected to global ipfs.)

Comment (Ellenn-A) [reply definitions/schemaDefinition.json:1]: Another attempt: 


### Schema Definition

The repo contains 'definitions' folder with a schema definition json which can be imported into ts files like so:

```
import _schema from './definitions/schemaDefinition.json'
const schema = _schema as AnonCredsSchema
```

The schema gives us a blueprint for providing data for a particular check. It requires these fields:

- checkName => what sort of check this is (e.g. NASDAQ, Rev >500, Production Capacity etc)
- companyName => name of the company that is being checked (max 200 characters)
- companiesHouseNumber => unique identifier for companies provided by Companiess HOuse (max 8 characters - alphanumeric)
- issueDate => date when this check was issued (timestamp)
- expiryDate => date when this check expires (timestamp)

The schema definition can be posted to `/schemas` to register it. Upon a successful call a response is returned with an 'id' property which refers to this schema and can be used to refer to the schema when creating a new credential definition like so:

```
{
  "tag": "myTestDefinition",
  "schemaId": "ipfs://example",
  "issuerId": "did:key:exampleDidKey"
}
```

A credential definition can then be used to issue a credential which contains both information about an issuer of a credential and the check itself.
(Note: when registering a schema one needs to have an instance of ipfs running or be connected to global ipfs.)

Comment (jonmattgray) [reply definitions/schemaDefinition.json:1]: yeah that's great 👍

**Review by jonmattgray — APPROVED** [review]

**Review by dblane-digicatapult — APPROVED** [review]

**Pull request merged**